test/contracts/MgnToken.sol

```diff
@@ -0,0 +1,177 @@
+    // @param   _minter the minter of the Magnolia tokens, should be the DX-proxy
+    function updateMinter(
+        address _minter
+    )
+        public
+    {
+        require(msg.sender == owner);
+        require(_minter != address(0));
+
+        minter = _minter;
+    }
+
+    // @dev the intention is to set the owner as the DX-proxy, once it is deployed
+    // Then only an update of the DX-proxy contract after a 30 days delay could change the minter again.
+    function updateOwner(   
+        address _owner
+    )
+        public
+    {
+        require(msg.sender == owner);
+        require(_owner != address(0));
+        owner = _owner;
+    }
+
+    function mintTokens(
+        address user,
+        uint amount
+    )
+        public
+    {
+        require(msg.sender == minter);
+
+        lockedTokenBalances[user] = add(lockedTokenBalances[user], amount);
+        totalSupply_ = add(totalSupply_, amount);
+    }
+
+    /// @dev Lock Token
+    function lockTokens (
+        uint amount
+    )
+        public
+        returns (uint totalAmountLocked)
+    {
+        // Adjust amount by balance
+        amount = min(amount, balances[msg.sender]);
+        
+        // Update state variables
+        balances[msg.sender] = sub(balances[msg.sender], amount);
+        lockedTokenBalances[msg.sender] = add(lockedTokenBalances[msg.sender], amount);
+
+        // Get return variable
+        totalAmountLocked = lockedTokenBalances[msg.sender];
+    }
+
+    function unlockTokens (
+        uint amount
+    )
+        public
+        returns (uint totalAmountUnlocked, uint withdrawalTime)
+    {
+        // Adjust amount by locked balances
+        amount = min(amount, lockedTokenBalances[msg.sender]);
+
+        if (amount > 0) {
+            // Update state variables
+            lockedTokenBalances[msg.sender] = sub(lockedTokenBalances[msg.sender], amount);
+            unlockedTokens[msg.sender].amountUnlocked =  add(unlockedTokens[msg.sender].amountUnlocked, amount);
+            unlockedTokens[msg.sender].withdrawalTime = now + 24 hours;
+        }
+
+        // Get return variables
+        totalAmountUnlocked = unlockedTokens[msg.sender].amountUnlocked;
+        withdrawalTime = unlockedTokens[msg.sender].withdrawalTime;
+    }
+
+    function withdrawUnlockedTokens()
+        public
+    {
+        require(unlockedTokens[msg.sender].withdrawalTime < now);
+        balances[msg.sender] = add(balances[msg.sender], unlockedTokens[msg.sender].amountUnlocked);
+        unlockedTokens[msg.sender].amountUnlocked = 0;
+    }
+
+    function min(uint a, uint b) 
+        public
+        pure
+        returns (uint)
+    {
+        if (a < b) {
+            return a;
+        } else {
+            return b;
+        }
+    }
+        /// @dev Returns whether an add operation causes an overflow
+    /// @param a First addend
+    /// @param b Second addend
+    /// @return Did no overflow occur?
+    function safeToAdd(uint a, uint b)
```

Comment:
why not use zeppelin safe math ?

Reply:
Cause that's a copy of the original MgnToken. We wanted to use the same logic. Just for testing.

We do plan to change to Safe Math at some point, but is not the porpoise of this project.

Reply:
Anyways, I just delete the file cause we use it to do some tests. But now we use the original MGN